pkg/api/controller.go

```diff
@@ -126,14 +127,24 @@ func (c *Controller) DeleteObjects(w http.ResponseWriter, r *http.Request, body
 				StatusCode: http.StatusUnauthorized,
 				Message:    http.StatusText(http.StatusUnauthorized),
 			})
-			continue
+		} else {
+			pathsToDelete = append(pathsToDelete, objectPath)
 		}
+	}
 
+	// batch delete the entries we allow to delete
+	delErr := c.Catalog.DeleteEntries(ctx, repository, branch, pathsToDelete)
+	delErrs := graveler.NewMapDeleteErrors(delErr)
+	for _, objectPath := range pathsToDelete {
```

Comment:
Why are you iterating the pathsToDelete, rather than the actual errors?
What should happen if all deletions in batch are successful?

Reply:
At the first part of the function we verified authorization - each file that we do not have access we added an error or added to 'pathsToDelete'.
The list 'pathsToDelete' represents all the files we need to delete (errors will include why we can't delete the rest).
Part of the protocol we need to report (optional - quiet flag) which files we deleted or not.
So going over the files it is to either report a successful delete or the specific/general error why it is not.